### PR TITLE
Lang. + formatting fixes in Security overview

### PR DESCRIPTION
**Pull request by rkratky**
- lang. fixes
- mark-up fixes
- couple of fixes outside of the security topic to make checks pass

I also suggest renaming the article to "Subiquity security overview" to make it more specific. I didn't rename the file itself to make it easier to see the suggested edits, but if accepted, I'd suggest renaming the file, too.

### CODE REVIEW

File: doc/explanation/security-overview.rst [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
 
-Ubuntu-server ISO has listening by default with a random password
------------------------------------------------------------------
+Ubuntu-server ISO is listening by default with a random password
+----------------------------------------------------------------
 
-The Ubuntu Server ISO offers SSH access to the installation system, to
-facilitate installs which need to start over a minimal serial line that may not
+The Ubuntu Server ISO offers SSH access to the installation system to
+facilitate installations that need to start over a minimal serial line, and that may not
```
Comment (dbungert) [inline doc/explanation/security-overview.rst]: This is intending to talk about two use cases
1. serial access, which may or may not be rich enough to run the installer in any capacity
2. other users who for their own reasons would prefer SSH access - richer language support is one reason

**Review by dbungert — APPROVED** [review]
This is all great, thanks.  I added one point of clarity which might change the wording, in the SSH text.

File: doc/explanation/security-overview.rst [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 be rich enough to run the installer user interface.  In that case, the SSH
 access information is printed on that serial line.
```
Comment (rkratky) [inline doc/explanation/security-overview.rst]:
```suggestion
facilitate the following installation use cases:

 * The installation needs to start over a minimal serial line, which may not
be capable of running the installer user interface; in that case, the SSH
access information is printed on that serial line.

* The user prefers using SSH access to interact with the installer interface
(for example, for richer language support).
```

File: doc/explanation/security-overview.rst (hunk 1 above)
Comment (rkratky) [reply doc/explanation/security-overview.rst]: Thanks, @dbungert. I modified the wording to reflect your note. I'm curious though: how does SSH access offer richer lang. support?

Comment (dbungert) [reply doc/explanation/security-overview.rst]: If you're physically present at a linux non-GUI console, that console lacks support for some languages that we can support in a modern GUI-based unicode terminal.

https://github.com/canonical/subiquity/blob/main/languagelist has the list, the first column has the value `console` or `ssh` which shows the languages that can be shown in all environments (`console`) or only in the fuller unicode experience (`ssh`).

Comment (rkratky) [reply doc/explanation/security-overview.rst]: Thank you!

**Pull request merged**